flink-runtime/src/main/java/org/apache/flink/runtime/rpc/resourcemanager/ResourceManager.java

```diff
@@ -52,18 +61,45 @@
 public class ResourceManager extends RpcEndpoint<ResourceManagerGateway> {
 	private final ExecutionContext executionContext;
 	private final Map<JobMasterGateway, InstanceID> jobMasterGateways;
+	private final Map<ResourceID, TaskExecutorGateway> taskExecutorGateways;
+	private final Map<ResourceID, ResourceManagerToTaskExecutorHeartbeatScheduler> heartbeatSchedulers;
+	private final LeaderElectionService leaderElectionService;
+	private UUID leaderSessionID;
+	// TODO private final SlotManager slotManager;
 
-	public ResourceManager(RpcService rpcService, ExecutorService executorService) {
+	public ResourceManager(RpcService rpcService, ExecutorService executorService, LeaderElectionService leaderElectionService) {
```

Comment:
Cool. maybe i could add a method like "LeaderElectionService getResourceManagerLeaderElectionService() throws Exception; " in HighAvailabilityServices class?